docs/_chapters/230-manifest-annotations.md

```diff
@@ -1,136 +1,284 @@
 
-### RequireCapability
+The [`@Requirement`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Requirement) and [`@Capability`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Capability) annotations were designed to address this issue. These annotations can be used to create [*custom bundle annotations*](#custom-bundle-annotations), described later on. Let's discuss the DS example.
 
-The `@RequireCapability` annotation creates a clause in the Require-Capability manifest header. The annotation has the following fields (for more details consult the Javadoc):
+Recent DS specifications require implementations to provide the following capability:
 
-* `value` – (String) Anything in the value field is appended after the calculated header.
-* `ns` – (String) The namespace of the requirement
-* `effective` – (String default `resolve`) The effective time
-* `filter` – The filter directive. There is a handy macro [${frange;<version>][1] that turns a version into a filter expression for a filter range.  
-* `resolution` – (`mandatory` | `optional`) The resolution of the requirement.
+```
+Provide-Capability: osgi.extender;
+    osgi.extender="osgi.component";
+    version:Version="x";
+    uses:="org.osgi.service.component"
+```
 
-### ProvideCapability
+While this provides a capability that can be required, we need a requirement to be generated from client code that uses DS. Enter recent versions of DS annotations which are meta-annotated with `@RequireServiceComponentRuntime`, a _custom bundle annotation_ which is specified as:
 
-The `@ProvideCapability` annotation creates a clause in the Provide-Capability manifest header. The annotation has the following fields (for more details consult the Javadoc):
+```java
+@Requirement(
+    namespace = ExtenderNamespace.EXTENDER_NAMESPACE,
+	name = ComponentConstants.COMPONENT_CAPABILITY_NAME,
+	version = ComponentConstants.COMPONENT_SPECIFICATION_VERSION)
+public @interface RequireServiceComponentRuntime { }
+```
 
-* `value` – (String) Anything in the value field is appended after the calculated header.
-* `ns` – (String) The namespace of the requirement
-* `effective` – (String default `resolve`) The effective time
-* `name` – (String) This creates a `<namespace>=<name>` attribute, the common convention to set the primary name attribute of a capability in most OSGi namespaces.
-* `version` – (String) Set the version of the capability.
-* `uses` – (String[]) Package names that are used by this capability and require the same class loader
-* `mandatory` – (String[]) List of mandatory attributes
+When you write a DS component using `@Component`
```

Comment:
I think this is missing a step - the doc doesn't say that the `@Component` annotation is meta-annotated with `@RequireServiceComponentRuntime` which means it is unclear why `@Component` causes the header generation

Reply:
I tried to imply this via the code comment, but I can make it more explicit.